File: HackerRank-10-Days-Statistics/Day 4/Day_4_Binomial_Distribution_I.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include<bits/stdc++.h>
using namespace std;


int factorial(int n){
    if(n==0 || n==1)
        return 1;
    return n*factorial(n-1);
}

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */  
    float a,b;
    float summation;
    cin>>a>>b;
    float p_b=a/(a+b);
    //float p_g=b/(a+b);
 
    
    for(int j=3;j<=6;j++){
        int n_c_r = factorial(6)/(factorial(j)*factorial(6-j));
        double powers =  pow(p_b,j)*pow(1-p_b,6-j);
        summation+= n_c_r * powers;
    }
    
    cout<<fixed<<setprecision(3)<<summation<<endl;
    //fixed<<setprecision(3)
     
    return 0;
}

File: HackerRank-10-Days-Statistics/Day 4/Day_4_Binomial_Distribution_I.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
a,b=list(map(float,input().strip().split()))

p_b=a/(a+b)

def factorial(n):
    if(n==0 or n==1):
        return 1
    return n*factorial(n-1)

summation=0

for i in range(3,7):
    n_C_r=factorial(6)/(factorial(i)*factorial(6-i))
    powers=(p_b**i)*((1-p_b)**(6-i))
    summation+=n_C_r*powers
    
print(round(summation,3))